fitbenchmarking/fitbenchmarking
# 736 scipy support for bounds

#### Description of Work

Fixes [#736]

Adds support for bounds to SciPy and SciPy LS controllers. Not all SciPy minimizers accept bounds, so in this case when a problem with parameter_ranges is used, the problem will not be run and results will appear as inf in tables (with an error flag explaining that the solver doesn't support bounded problems).

Tests have also been added to check that for those minimizers that accept bounds, they are being respected for bounded problems.

#### Testing Instructions

1. Run Fitbenchmarking with sasview examples (these problems have parameter bounds) with scipy and scipy_ls fitting software selected - check that minimizers that don't support bounds give inf results

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
